### PR TITLE
Speed up requirejs tests

### PR DESCRIPTION
**Pull request by orangejenny**
[A recent job](https://travis-ci.org/dimagi/commcare-hq/jobs/565956236) says setup for `test_files_match_modules` takes 20 sec and running takes 10, but it's about 4 times as fast locally. Times in commit messages are based on local testing.

### CODE REVIEW

**Review by gbova — APPROVED** [review]

File: corehq/apps/hqwebapp/tests/test_requirejs.py [hunk 1]
```diff
@@ -21,6 +21,18 @@ def _run_jobs(cls, files, processor):
...
+    @classmethod
+    def _hqDefine_line(cls, filename):
+        hqDefine_line = None
+        with open(filename, 'r') as f:
+            line = f.readline()
+            while line and not hqDefine_line:
+                if re.match(r'^\s*hqDefine', line):
+                    hqDefine_line = line
+                line = f.readline()
+        return hqDefine_line
```
Comment (millerdev) [inline corehq/apps/hqwebapp/tests/test_requirejs.py]: nit: what do you think of using a for-loop here? It also might be slightly more efficient to pre-compile the regex (although I think the re module does keep a cache of the last 100 expressions its seen, so probably negligible):
```py
def _hqDefine_line(cls, filename):
    regex = re.compile(r'^\s*hqDefine')
    with open(filename, 'r') as f:
        for line in f:
            if regex.match(line):
                return line
    return None
```

File: corehq/apps/hqwebapp/tests/test_requirejs.py [hunk 2]
```diff
@@ -77,15 +87,16 @@ def test_requirejs_disallows_hqimport(self):
...
-            for line in [b.decode('utf-8') for b in out.split(b"\n")]:
-                if line:
-                    match = re.search(r'hqImport\([\'"]([^\'"]*)[\'"]', line)
-                    if match:
-                        errors.append("{} imported in {}".format(match.group(1), filename))
-                    else:
-                        errors.append("hqImport found in {}: {}".format(filename, line.strip()))
+            with open(filename, 'r') as f:
+                line = f.readline()
+                while line:
```
Comment (millerdev) [inline corehq/apps/hqwebapp/tests/test_requirejs.py]: nit: use of `f.readline()` (also below on line 99) and uncompiled regexes. Also, literal string search is way faster than a regex.

Suggested changes:
```py
hqImport = re.compile(r'hqImport\([\'"]([^\'"]*)[\'"]')
for line in f:
    if 'hqImport' in line:
        match = hqImport.search(line)
```

**Review by millerdev — APPROVED** [review]

File: corehq/apps/hqwebapp/tests/test_requirejs.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/hqwebapp/tests/test_requirejs.py]: Yes, that's much cleaner, thanks. dd4f9d760ca

File: corehq/apps/hqwebapp/tests/test_requirejs.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/hqwebapp/tests/test_requirejs.py]: dd4f9d760ca

**Review by millerdev — APPROVED** [review]

File: corehq/apps/hqwebapp/tests/test_requirejs.py [hunk 3]
```diff
@@ -21,29 +21,41 @@ def _run_jobs(cls, files, processor):
...
+        design_pattern = re.compile(r'/_design/')
+        couchapps_pattern = re.compile(r'couchapps')
+        vellum_pattern = re.compile(r'/vellum/')
         cls.js_files = [f for f in [b.decode('utf-8') for b in out.split(b"\n")] if f
-                    and not re.search(r'/_design/', f)
-                    and not re.search(r'couchapps', f)
-                    and not re.search(r'/vellum/', f)]
+                    and not design_pattern.search(f)
+                    and not couchapps_pattern.search(f)
+                    and not vellum_pattern.search(f)]
```
Comment (millerdev) [inline corehq/apps/hqwebapp/tests/test_requirejs.py]: nit: I think all of these could be changed to the form `literal not in f`, which should be more efficient than using regular expressions. For example, this one would be
```suggestion
                    and '/vellum/' not in f]
```

Comment (orangejenny) [reply corehq/apps/hqwebapp/tests/test_requirejs.py]: 616b974dab0f08d02124642744df303cacff8263

**Pull request merged**